src/panels/KaitoModelsPanel.ts

```diff
@@ -96,21 +100,102 @@ export class KaitoModelsPanelDataProvider implements PanelDataProvider<"kaitoMod
     nullIsFalse(value: boolean | null): boolean {
         return value ?? false;
     }
+    parseGPU(gpuRequirement: string): [string, number] {
+        const regex = /^Standard_NC(\d+)(ads_A100_v4|s_v3)$/;
+        const match = gpuRequirement.match(regex);
+        if (match) {
+            const cpus = parseInt(match[1], 10);
+            let family: string;
+            switch (match[2]) {
+                case "ads_A100_v4":
+                    family = "StandardNCADSA100v4Family";
+                    break;
+                case "s_v3":
+                    family = "standardNCSv3Family";
+                    break;
+                default:
+                    throw new Error("Unknown gpu format");
+            }
+
+            return [family, cpus];
+        } else {
+            throw new Error("Unknown gpu format");
+        }
+    }
 
-    private async handleDeployKaitoRequest(model: string, yaml: string, webview: MessageSink<ToWebViewMsgDef>) {
+    private async handleDeployKaitoRequest(
+        model: string,
+        yaml: string,
+        gpu: string,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
         this.cancelToken = false;
+        if (this.checkingGPU) {
+            return;
+        }
         try {
+            this.checkingGPU = true;
```

Comment:
💡 Highly recommended: Can you please comment this function with the intent and the usage of various things like setting `this.checkingGPU = true` 

something like:

\** 

Intent of this function is to handleDeploy with few default values set ...... <Explanation for latter understanding>

**/